site/src/main/java/uk/nhs/digital/ps/beans/Publication.java

```diff
@@ -66,43 +68,67 @@ public RestrictableDate getNominalPublicationDate() {
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:CoverageStart")
     public Calendar getCoverageStart() {
-        return getProperty("publicationsystem:CoverageStart");
+        return getPermittedProperty("publicationsystem:CoverageStart");
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:CoverageEnd")
     public Calendar getCoverageEnd() {
-        return getProperty("publicationsystem:CoverageEnd");
+        return getPermittedProperty("publicationsystem:CoverageEnd");
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:GeographicCoverage")
     public String getGeographicCoverage() {
-        return getProperty("publicationsystem:GeographicCoverage");
+        return getPermittedProperty("publicationsystem:GeographicCoverage");
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:Granularity")
     public String[] getGranularity() {
-        return getProperty("publicationsystem:Granularity");
+        return getPermittedProperty("publicationsystem:Granularity");
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:AdministrativeSources")
     public String getAdministrativeSources() {
-        return getProperty("publicationsystem:AdministrativeSources");
+        return getPermittedProperty("publicationsystem:AdministrativeSources");
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:Title")
     public String getTitle() {
         return getProperty("publicationsystem:Title");
     }
 
+    public boolean isPubliclyAccessible() {
+        return getProperty("publicationsystem:PubliclyAccessible");
+    }
+
     @HippoEssentialsGenerated(internalName = "publicationsystem:RelatedLinks")
     public List<Relatedlink> getRelatedLinks() {
-        return getChildBeansByName("publicationsystem:RelatedLinks",
-                Relatedlink.class);
+        return getPermittedChildBeansByName("publicationsystem:RelatedLinks", Relatedlink.class);
     }
 
     @HippoEssentialsGenerated(internalName = "publicationsystem:attachments")
     public List<HippoResourceBean> getAttachments() {
-        return getChildBeansByName("publicationsystem:attachments", HippoResourceBean.class);
+        return getPermittedChildBeansByName("publicationsystem:attachments", HippoResourceBean.class);
+    }
+
+    private <T extends HippoBean> List<T> getPermittedChildBeansByName(final String propertyName, final
+    Class<T> beanMappingClass) {
+        assertPropertyAccessible(propertyName);
+
+        return getChildBeansByName(propertyName, beanMappingClass);
+    }
+
+    private <T> T getPermittedProperty(final String propertyName) {
```

Comment:
I find this naming (and that for attachments) a bit awkward. What about "getPropertyIfPublished"?

Reply:
Agreed. I'll rename to `getPropertyIfPermitted` and apply to all getters.

Reply:
Done.